Sources/StreamChat/Database/DTOs/AttachmentDTO.swift

```diff
@@ -41,6 +41,16 @@ class AttachmentDTO: NSManagedObject {
     
     @NSManaged var message: MessageDTO
     @NSManaged var channel: ChannelDTO
+
+    override func willSave() {
+        super.willSave()
+
+        // When attachment changed, we need to propagate this change up to holding message
+        if hasPersistentChangedValues, !message.hasChanges {
+            // this will not change object, but mark it as dirty, triggering updates
+            message.id = message.id
```

Comment:
How come this change is not removed by https://github.com/GetStream/stream-chat-swift/pull/802/commits/df7983b65dac26188897ea48caad9bf8cf4d0b03 ?

Reply:
`willSave` is called inside `context.save()`, so our "cleanup" logic is already performed.

Reply:
Gotcha 👍